common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -1527,65 +1527,55 @@ bool ap_wlan_hal_dwpal::read_acs_report()
 
 bool ap_wlan_hal_dwpal::read_supported_channels()
 {
-    char *reply = nullptr;
-    if (!dwpal_send_cmd("GET_HW_FEATURES", &reply)) {
-        LOG(ERROR) << __func__ << " failed";
-        return false;
-    }
+    LOG(TRACE) << __func__ << " for interface: " << get_radio_info().iface_name;
 
-    size_t replyLen = strnlen(reply, HOSTAPD_TO_DWPAL_MSG_LENGTH);
-    for (size_t i = 0; i < replyLen; i++) {
-        if (reply[i] == ')') {
-            reply[i] = ' ';
-        }
+    // Clear the supported channels vector
+    m_radio_info.supported_channels.clear();
+    // Resize the supported channels vector
+    m_radio_info.supported_channels.resize(MAX_SUPPORTED_20M_CHANNELS);
```

Comment:
It would feel more comfortable to use`MAX_SUPPORTED_CHANNELS`

Reply:
The value of `MAX_SUPPORTED_20M_CHANNELS` is defined to `beerocks::message::SUPPORTED_CHANNELS_LENGTH` which is the current size to use, since that is the size allocated in the outgoing message.
On the other hand, there is a `MAX_SUPPORTED_CHANNELS`, which is set to `(MAX_SUPPORTED_20M_CHANNELS*4)`
Which is larger then the amount I wish to allocate

Reply:
by map i presume you are talking about the AP Radio Basic Capabilities
see #686
I plan on adding the AP Radio Basic Capabilities "map" in the next PR

and as for the vector,
Currently the supported_channels are stored in a vector since they have no regard to the bandwidth, they are translated later on (`ap_manager_thread` & `son_slave_thread`)

Reply:
This is wrong - supported channels must contain channel and bandwidth. Please use the same data structure defined in nl80211_client.

Reply:
With the new changes the supported channels have the channel number, the supported bandwidth, supported tx power & whether the channel is dfs or not

The supported channels are stored in a vector, and later on sent to the son_slave_thread where we build with them the ap radio basic capabilities